File: backend/src/models/quota.model.js
const db = require("../database/db");

const createQuota = (data, callback) => {
  const sql = `
    INSERT INTO tbd_cuotas
      (reservacion_id, mes, anio, tipo_pago, monto_calculado, estado, fecha_vencimiento)
    VALUES (?, ?, ?, ?, ?, ?, ?)
  `;

  const values = [
    data.reservacion_id,
    data.mes,
    data.anio,
    data.tipo_pago || "mensual",
    data.monto_calculado,
    data.estado || "pendiente",
    data.fecha_vencimiento || null,
  ];

  db.query(sql, values, callback);
};

const findQuotaByReservationPeriod = (reservacion_id, mes, anio, callback) => {
  const sql = `
    SELECT *
    FROM tbd_cuotas
    WHERE reservacion_id = ? AND mes = ? AND anio = ?
    LIMIT 1
  `;

  db.query(sql, [reservacion_id, mes, anio], callback);
};

const updateQuotaStatusByReservationPeriod = (reservacion_id, mes, anio, estado, callback) => {
  const sql = `
    UPDATE tbd_cuotas
    SET estado = ?
    WHERE reservacion_id = ? AND mes = ? AND anio = ?
  `;

  db.query(sql, [estado, reservacion_id, mes, anio], callback);
};

module.exports = {
  createQuota,
  findQuotaByReservationPeriod,
  updateQuotaStatusByReservationPeriod,
};

File: API/models/quota_model.py
from sqlalchemy import Column, Integer, String, Float
from config.database import Base

class QuotaModel(Base):
    __tablename__ = "tbd_cuotas"
    
    id = Column(Integer, primary_key=True, index=True)
    reservacion_id = Column(Integer, index=True)
    mes = Column(Integer)
    anio = Column(Integer)
    tipo_pago = Column(String(50))
    monto_calculado = Column(Float)
    estado = Column(String(50), default="pendiente")
    fecha_vencimiento = Column(String(50))
